Source/WebCore/rendering/RenderTable.cpp

```diff
@@ -620,14 +620,12 @@ void RenderTable::layout()
     }
 
     bool paginated = layoutState && layoutState->isPaginated();
-    if (sectionMoved && paginated) {
+    if (sectionCount && sectionMoved && paginated) {
         // FIXME: Table layout should always stabilize even when section moves (see webkit.org/b/174412).
-        if (m_recursiveSectionMovedWithPaginationLevel < sectionCount) {
-            SetForScope recursiveSectionMovedWithPaginationLevel(m_recursiveSectionMovedWithPaginationLevel, m_recursiveSectionMovedWithPaginationLevel + 1);
-            markForPaginationRelayoutIfNeeded();
-            layoutIfNeeded();
-        } else
-            ASSERT_NOT_REACHED();
+        ASSERT(m_recursiveSectionMovedWithPaginationLevel < sectionCount);
```

Comment:
The old code was safer because if this assertion was not met, the code would be skipped. Was there a compelling reason to remove the if statement?

Reply:
I thought the code is identical but you're right it would be different for release builds.